consul/tests/test.bal

```diff
@@ -54,7 +54,7 @@ function testGetService() {
         io:println(<string>serviceDetails.detail().message);
         test:assertFail(msg = <string>serviceDetails.detail().message);
     } else {
-        test:assertNotEquals(serviceDetails, null, msg = "Failed to call getService()");
+        test:assertNotEquals(serviceDetails, (), msg = "Failed to call getService()");
```

Comment:
`getService()` returns `CatalogService[]|error` right? So it can never be `()`, and this test never fails. 

If what we want to assert is than an error is not returned, we could just do an `test:assertTrue(serviceDetails is CatalogService[])` instead of these if-else blocks. But shouldn't we be doing further assertions when an array is returned, based on the input? Let's create an issue for this maybe, to improve the tests. IINM, this is also the case for some other connectors as well.